openstack-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1004 @@
+              options: {
+                down: {
+                  color: 'red',
+                  index: 1,
+                  text: 'Down',
+                },
+                up: {
+                  color: 'green',
+                  index: 0,
+                  text: 'Up',
+                },
+              },
+            }
+          ),
+        ),
+      ])
+      + table.queryOptions.withTransformationsMixin([
+        {
+          id: 'joinByField',
+          options: {
+            byField: 'Time',
+            mode: 'outer',
+          },
+        },
+        {
+          id: 'organize',
+          options: {
+            excludeByName: {
+              Time: true,
+              Value: true,
+              __name__: true,
+              hostname: true,
+              job: true,
+            },
+            includeByName: {},
+            indexByName: {
+              Time: 6,
+              Value: 9,
+              __name__: 7,
+              adminState: 3,
+              hostname: 5,
+              id: 4,
+              instance: 1,
+              job: 8,
+              service: 0,
+              zone: 2,
+            },
+            renameByName: {
+              Time: '',
+              adminState: 'Admin state',
+              id: 'ID',
+              instance: 'Instance',
+              service: 'Service',
+              zone: 'Zone',
+            },
+          },
+        },
+      ]),
+
+    volumes:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Volumes',
+        targets=[t.volumes],
+        description='The number of volumes managed by Cinder.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    volumeStatus:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Volume status',
+        targets=[t.volumeErrorStatus, t.volumeNonErrorStatus],
+        description='The current status of volumes in Cinder.',
+      )
+      + timeSeries.options.legend.withPlacement('right')
+      + timeSeries.fieldConfig.defaults.custom.withStackingMixin({
+        group: 'A',
+        mode: 'normal',
+      })
+      + timeSeries.standardOptions.withDecimals(0),
+
+    volumeUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
+        'Volume usage',
+        targets=[t.volumeUsage],
+        description='The percent of volume storage in use for Cinder.',
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    backupUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
+        'Backup usage',
+        targets=[t.backupUsage],
+        description='The percent of backup storage in use for Cinder.',
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    poolUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
```

Comment:
This can be changed from generic into one of the storage panels from the common lib.

Reply:
I don't see any of the common lib panels that fit this one. Do you have a specific one in mind?

Reply:
Yea sorry, instead of storage its called memory in the commonlib
<img width="767" alt="image" src="https://github.com/grafana/jsonnet-libs/assets/38274348/ff09d750-2629-4b47-9f3c-fd287e3c9a89">